server/backend/api/app/crud/crud_user.py

```diff
@@ -246,9 +247,142 @@ def fetch_initial_requester_info(
+                if user.user_guid:
+                    # if found business bceid user by user_guid, update username if necessary
+                    search_result = idim_proxy_service.search_business_bceid(
+                        schemas.IdimProxyBceidSearchParam(
+                            **{
+                                "searchUserBy": IdimSearchUserParamType.USER_GUID,
+                                "searchValue": user.user_guid,
+                            }
+                        )
+                    )
+                    properties_to_update = {
+                        models.FamUser.user_name: search_result.get("userId"),
+                    }
+
+                else:
+                    # if user has no user_guid in our database, find by user_name and add user_guid to database
+                    search_result = idim_proxy_service.search_business_bceid(
+                        schemas.IdimProxyBceidSearchParam(
+                            **{
+                                "searchUserBy": IdimSearchUserParamType.USER_ID,
+                                "searchValue": user.user_name,
+                            }
+                        )
+                    )
+                    properties_to_update = {
+                        models.FamUser.user_guid: search_result.get("guid"),
+                    }
+            else:
+                # ignore bc service card users
+                ignored_user_list.append(user.user_id)
+                LOGGER.debug(f"Updating information for user {user.user_name} is ignored because we only focus on IDIR and Business BCeID")
+                continue
+
+            # Update various target_user fields from idim search if exists
+            if search_result and search_result.get("found"):
+                properties_to_update = {
+                    **properties_to_update,
+                    models.FamUser.first_name: search_result.get("firstName"),
+                    models.FamUser.last_name: search_result.get("lastName"),
+                    models.FamUser.email: search_result.get("email"),
+                    models.FamUser.business_guid: search_result.get("businessGuid"),
+                }
+
+                update(db, user.user_id, properties_to_update, requester.cognito_user_id)
+                LOGGER.debug(f"Updating information for user {user.user_name} is done")
+                success_user_list.append(user.user_id)
+            else:
+                LOGGER.debug(
+                    f"Invalid request, cannot find user {user.user_name} {user.user_guid} with user type {user.user_type_code}"
+                )
+                failed_user_list.append(user.user_id)
+
+        except Exception as e:
+            LOGGER.debug(f"Failed to update user info: {e}")
+            failed_user_list.append(user.user_id)
+
+    return schemas.FamUserUpdateResponse(
+        **{
+            "total_users_count": total_users_count,
+            "current_page": page,
```

Comment:
if `not use_pagination`, do you still intend to give it a value for "current_page" as 1?

Reply:
I'm actually fine with it... do you feel confuse?

Reply:
I only wonder your intention since at one hand use_pagination is False, but on the other hand it returns a current_page. I think it is fine. No big deal.